src/test/java/io/jenkins/plugins/folderauth/FolderAuthorizationWebAPITest.java

```diff
@@ -0,0 +1,306 @@
+                apiPath = "folder-auth/addFolderRole";
+                break;
+            case AGENT:
+                apiPath = "folder-auth/addAgentRole";
+                break;
+            default:
+                throw new IllegalArgumentException("Unexpected role type");
+        }
+        URL apiURL = new URL(jenkinsRule.jenkins.getRootUrl() + apiPath);
+        WebRequest request = new WebRequest(apiURL, HttpMethod.POST);
+        request.setRequestBody(jsonString);
+        request.setAdditionalHeader("Content-Type", "application/json");
+        Page page = webClient.getPage(request);
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+    }
+
+    /**
+     * Util method to verify that the role exists
+     * @param roleType Type of role
+     * @param roleMap Map object that represents the role
+     * @throws NoSuchMethodException
+     * @throws InvocationTargetException
+     * @throws IllegalAccessException
+     */
+    private void verifyThatRoleExists(RoleType roleType, Map<String, Object> roleMap)
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        AuthorizationStrategy a = jenkinsRule.jenkins.getAuthorizationStrategy();
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) a;
+        boolean found = false;
+        // Using reflection to get method
+        String getRoleMethodString = "get" + StringUtils.capitalize(roleType.name().toLowerCase()  + "Roles");
+        Method getRoleMethod = strategy.getClass().getMethod(getRoleMethodString);
+        Set<AbstractRole> roleSet = (Set) getRoleMethod.invoke(strategy);
+        for (AbstractRole role : roleSet) {
+            if (role.getName().equals(roleMap.get("name"))) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
+    }
+
+    /**
+     * Util method to verify that a user has been assigned to a particular role
+     * @param roleType Type of role
+     * @param sid User ID
+     * @param roleName Name of the role
+     */
+    private void verifyUserAssignedToRole(RoleType roleType, String sid, String roleName) {
+
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) jenkinsRule.jenkins.getAuthorizationStrategy();
+        boolean assigned = false;
+        Set<? extends AbstractRole> roles;
+        switch (roleType) {
+            case GLOBAL:
+                roles = strategy.getGlobalRoles();
+                break;
+            case FOLDER:
+                roles = strategy.getFolderRoles();
+                break;
+            case AGENT:
+                roles = strategy.getAgentRoles();
+                break;
+            default:
+                throw new IllegalArgumentException("Unexpected role. Expecting GLOBAL, FOLDER or AGENT");
+        }
+
+        for (AbstractRole folderRole : roles) {
```

Comment:
Maybe change `folderRole` to just `role`
```suggestion
        for (AbstractRole role: roles) {
```